app/lib/package/models.dart

```diff
@@ -454,6 +454,22 @@ class PackageVersion extends db.ExpandoModel<String> {
     );
   }
 
+  /// Updates the current instance with the newly derived data.
+  /// Returns true if the current instance changed.
+  bool updateIfChanged({
+    required String? pubspecContentAsYaml,
+  }) {
+    var changed = false;
+    if (pubspecContentAsYaml != null) {
+      final newPubspec = Pubspec.fromYaml(pubspecContentAsYaml);
+      if (pubspec!.jsonString != newPubspec.jsonString) {
```

Comment:
Shouldn't we use a different comparison? Like deep-equals after parsing from `jsonString`?

Reply:
We could use deep comparison, but I don't think it will have much benefit over the current implementation.

The specific counts in the tests show us that if we feed the same `pubspec.yaml` file, we generate the same json string, and don't trigger an update with it. In practice the JSON string will keep the order of the keys.

We have similar comparison with JSON at least one more place in our codebase, so if we wanted to have deep copy, it could be used at multiple places, but I'd consider it a low-priority. I'll add a TODO for it.

Reply:
A todo is fine :D